lib/stream_data.ex

```diff
@@ -136,6 +157,11 @@ defmodule StreamData do
   end
 
   # TODO: should this be exposed?
+  # QUESTION: In Elixir, we would call this function filter_map,
+  # and have it return {:cont, value} | :skip or {:ok, value} | :error.
+  # Maybe we should standardize this return type and make this public?
+  # I would also likely call the underlying rose tree operation filter_map.
```

Comment:
@josevalim sounds good. I like `{:cont, term} | :skip`, or `{:keep, term} | :skip` more than `{:ok, term} | :error` because there's no "error" here. Wdyt?